app/controllers/representatives_controller.rb

```diff
@@ -0,0 +1,46 @@
+class RepresentativesController < ApplicationController
+  def index
+    @representatives = Representative.all
+  end
+
+  def show
+    @representative = Representative.find(params[:id])
+    @reviews = @representative.reviews
+    @comments = {}
+    @reviews.each do |review|
+      @comments[review.id] = review.comments
+    end
+  end
+
+  def new
+    @representative = Representative.new
+  end
+
+  def create
+    @representative = Representative.create(representative_params)
+    if @representative.save
+      redirect_to @representative, notice: "Representative added successfully"
+    else
+      flash[:alert] = @representative.errors.full_messages.join(', ')
+      render :new
+    end
+  end
+
+  def edit
```

Comment:
Generally you do not want to leave empty actions in your controller like this. Leave them out until they are implemented